### PR TITLE
feat: Create Collaborators rest api

### PR DESCRIPTION
**Pull request by nverbic**
Description:
Create REST API for Collaborators page.
Closes https://github.com/Women-Coding-Community/wcc-backend/issues/20

Changes:
Add collaboratorPage.json:
  - Using only page section of json in this first draft version
  - Remove the "details" section from the json definition since it is not recognized as a part of the Member class.
  - TODO: json file will contain data section (this will be the page section tested above) and metadata section (number of pages
    and pageSize - this is to be added back to this json file when CollaboratorPagePaginated  is implemented)
  - TODO: Define if details field is needed - it is working well without it

Add CollaboratorPage.java

Modify CmsService.java:
  - Add public CollaboratorPage getCollaborator()

Modify AboutController.java:
  - GET mapping for collaborators

Modify ApiResourcesFile.java:
  - Add COLLABORATOR constant for the json file

Screenshot (66)
Screenshot (68)
Screenshot (69)
Screenshot (74)

Added tests for Collaborator in the same way as there are for Team page. 
Added test for Ok response when the page is successfully loaded in the same way as it is for Footer page.
ALL tests passed

Screenshot (76)

Screenshot (78)

### CODE REVIEW

File: src/main/resources/collaboratorPage.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          "type": "desktop"
+        }
+      ],
+      "network": [
+        {
+          "type": "youtube",
+          "link": "https://www.linkedin.com"
+        }
+      ],
+      "memberType": "director"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:37]: Here you have to add the same collaborators we have in the current website. (No directors, leaders or evangelist)

File: src/main/java/com/wcc/platform/controller/AboutController.java [hunk 2]
```diff
@@ -27,4 +28,10 @@ public AboutController(CmsService service) {
     public ResponseEntity<TeamPage> getTeamPage() {
         return ResponseEntity.ok(cmsService.getTeam());
     }
+
+    @GetMapping("/collaborators")
+    @ApiOperation(value = "API to retrieve information about collaborators")
+    public ResponseEntity<CollaboratorPage> getCollaboratorPage() {
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/controller/AboutController.java:34]: Creating tests and this is good to go.

File: src/main/java/com/wcc/platform/service/CmsService.java [hunk 3]
```diff
@@ -35,4 +37,18 @@ public TeamPage getTeam() {
             throw new PlatformInternalException(e.getMessage(), e);
         }
     }
+
+    /**
+     * API to retrieve information about collaborators.
+     *
+     * @return Collaborators page content.
+     */
+    public CollaboratorPage getCollaborator() {
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/service/CmsService.java:46]: add test also for this.

File: src/main/resources/collaboratorPage.json (hunk 1 above)
Comment (dricazenck) [reply src/main/resources/collaboratorPage.json:37]: No, I don't think so. You can just add the current collaborators we have in our website.

Comment (nverbic) [reply src/main/resources/collaboratorPage.json:37]: @dricazenck Ok. I updated the MemberType enum with WEBSITE_COLLABORATOR and TEAM_COLLABORATOR items but in the current way of enum implementation the items are displayed on the page with the underscore as well. I tried to change the enum to use the text value like this: WEBSITE_COLLABORATOR("WEBSITE COLLABORATOR") but that requires then other changes in the deserializer() function. We can discuss this later maybe.

File: src/main/resources/collaboratorPage.json [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    "title": "Team",
+    "subtitle": "Collaborators",
+    "description": "Our collaborators are the backbone of our community, offering vital expertise and support that drive our initiatives forward and enhance our digital presence."
+  },
+  "contact": {
+    "title": "Contact us",
+    "links": [
+      {
+        "type": "email",
+        "link": "london@wcc.com"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:12]:
```suggestion
        "link": "london@womencodingcommunity.com"
```

File: src/test/java/com/wcc/platform/controller/AboutControllerTest.java [hunk 5]
```diff
@@ -49,4 +49,28 @@ void testInternalError() throws Exception {
                 .andExpect(jsonPath("$.message", is("internal error")))
                 .andExpect(jsonPath("$.details", is("uri=/api/cms/v1/team")));
     }
+
+    @Test
+    void testCollaboratorNotFound() throws Exception {
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/controller/AboutControllerTest.java:54]: It is missing the success case, when we should match the response with success case.

File: src/main/resources/collaboratorPage.json [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "linkedin",
+          "link": "https://www.linkedin.com/in/lilynguyendk"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/lilyndk"
+        }
+      ],
+      "memberType": "MEMBER"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:41]:
```suggestion
      "memberType": "COLLABORATOR"
```

Also please update the enum like this: 
``` 
public enum MemberType {
    DIRECTOR,
    LEADER,
    EVANGELIST,
    VOLUNTEER,
    MENTOR,
    MENTEE,
    MEMBER,
    SPEAKER,
    COLLABORATOR,
}
```

File: src/main/resources/collaboratorPage.json [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "linkedin",
+          "link": "https://www.linkedin.com/in/chloe-britland-whiting/"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/cribw27"
+        }
+      ],
+      "memberType": "MEMBER"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:63]:
```suggestion
      "memberType": "COLLABORATOR"
```

File: src/main/resources/collaboratorPage.json [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "linkedin",
+          "link": "https://www.linkedin.com/in/nehha/"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/Neha"
+        }
+      ],
+      "memberType": "MEMBER"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:85]:
```suggestion
      "memberType": "COLLABORATOR"
```

File: src/main/resources/collaboratorPage.json [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "linkedin",
+          "link": "https://www.linkedin.com/in/sophyo/"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/sssoz"
+        }
+      ],
+      "memberType": "MEMBER"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:107]:
```suggestion
      "memberType": "COLLABORATOR"
```

File: src/main/resources/collaboratorPage.json [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "linkedin",
+          "link": "https://www.linkedin.com/in/chiarapigoni/"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/kikilondon"
+        }
+      ],
+      "memberType": "MEMBER"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:129]:
```suggestion
      "memberType": "COLLABORATOR"
```

File: src/main/resources/collaboratorPage.json [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+        {
+          "type": "linkedin",
+          "link": "link: https://www.linkedin.com/in/nevena-verbič-83a16ab/"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/nverbic"
+        }
+      ],
+      "memberType": "MEMBER"
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:151]:
```suggestion
      "memberType": "COLLABORATOR"
```

File: src/main/resources/collaboratorPage.json [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+          "link": "https://www.linkedin.com/in/chloe-britland-whiting/"
+        },
+        {
+          "type": "github",
+          "link": "https://github.com/cribw27"
+        }
+      ],
+      "memberType": "MEMBER"
+    },
+    {
```
Comment (dricazenck) [inline src/main/resources/collaboratorPage.json:65]:
```suggestion
     {
      "fullName": "Joana Brito",
      "position": "Frontend Engineer",
      "images": [
        {
          "path": "image5.jpeg",
          "alt": "Collaborator profile image",
          "type": "desktop"
        }
      ],
      "network": [
        {
          "type": "linkedin",
          "link": "https://www.linkedin.com/in/joana-brito216"
        },
        {
          "type": "github",
          "link": "https://github.com/joanaBrit"
        }
      ],
      "memberType": "VOLUNTEER"
    },
    {
```

File: src/test/java/com/wcc/platform/controller/AboutControllerTest.java (hunk 5 above)
Comment (nverbic) [reply src/test/java/com/wcc/platform/controller/AboutControllerTest.java:54]: I will add the test during the day.

Comment (nverbic) [reply src/test/java/com/wcc/platform/controller/AboutControllerTest.java:65]: Test added

**Review by dricazenck — APPROVED** [review]

**Pull request merged**